### PR TITLE
[Ray Core] Fix the Paths in the Generated Monitoring Configs to Consider temp-dir

### LINKED ISSUE #41648: [Ray Core]  "/tmp/ray"  should not be hard coded for dashboard/observation.
### What happened + What you expected to happen

When I config the --temp-dir  for the ray start ,    the output does redirect to the new folder.   However, the  generated artifacts  found in the  /session/latest/metrics/  are still using the "/tmp/ray " .   Ideally all the places once using "/tmp/ray"  should be replaced with the new  temp-dir.

### Versions / Dependencies

ray 2.8 

### Reproduction script

As described above. 

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by MengjinYan**
## Why are these changes needed?
Today, when a user specifies the `--temp-dir` when starting ray, the paths in the default config files generated are still with the default root directory `/tmp/ray`. The issue is because the default config files are directly copied from the package when ray starts and the paths are all hardcoded.

This PR fix the issue by generating the corresponding config files with the corresponding path at runtime. 

Also:
* keep using the default path with the hard coded file in the `ray metrics launch-prometheus` command because the command can be executed with/without ray cluster started. The doc is updated accordingly to suggest the user to use manual mode to start prometheus is --temp-dir will be specified
* Updated the tests to:
  * add verification of the monitoring config generation for the case of specify temp-dir
  * add verification of the paths in the config content

## Related issue number

Closes #41648 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/metrics.md [hunk 1]
```diff
@@ -88,13 +94,17 @@ scrape_configs:
 - job_name: 'ray'
   file_sd_configs:
   - files:
-    - '/tmp/ray/prom_metrics_service_discovery.json'
+    - '/tmp/ray/prom_metrics_service_discovery.json' # or '{your_temp_path}/prom_metrics_service_discovery.json' if --temp-dir is specified
```
Comment (rynewang) [inline doc/source/cluster/metrics.md:97]: [nit] ${your_temp_path} the dollar sign

File: python/ray/dashboard/modules/metrics/install_and_start_prometheus.py [hunk 2]
```diff
@@ -74,12 +74,18 @@ def install_prometheus(file_path):
 
 def start_prometheus(prometheus_dir):
 
-    # Currently, Ray never modifies this config file, so we can just use the
-    # hardcoded path. (It just copies it to a more user-friendly location, in
-    # MetricsHead._create_default_prometheus_configs.)
-    # However, if in the future Ray ever modifies this file at runtime, we'll
-    # need to use the user-friendly location instead, and reload the config
-    # file after it's updated by Ray.
+    # The function assumes, to-be-prometheus-monitored ray cluster uses the default
```
Comment (rynewang) [inline python/ray/dashboard/modules/metrics/install_and_start_prometheus.py:77]: Nit:

```
# This function assumes the Ray cluster to be monitored by Prometheus uses the default 
# configuration with "/tmp/ray" as the root temporary directory.
# 
# This is to support the `ray metrics launch-prometheus` command, when a ray cluster
# is not started yet and we have no way to get a `--temp-dir` anywhere. So we choose to
# use a hardcoded default value.
```

**Review by rynewang — APPROVED** [review]

**Comment by rynewang** [comment]
@angelinalg please take a look at this - Mengjin fixed an issue about when users specify `--temp-dir`, the default grafana config no longer works. We added docs for those people about how to config their grafana.

File: doc/source/cluster/metrics.md [hunk 3]
```diff
@@ -22,6 +22,12 @@ For a quick demo, you can run Prometheus locally on your machine. Follow the qui
 
 ### Quickstart: Running Prometheus locally
 
+```{admonition} Note
+:class: note
+If you need to change the root temporary directory by using "--temp-dir" in your ray
```
Comment (angelinalg) [inline doc/source/cluster/metrics.md:27]:
```suggestion
If you need to change the root temporary directory by using "--temp-dir" in your Ray 
```

File: doc/source/cluster/metrics.md [hunk 4]
```diff
@@ -22,6 +22,12 @@ For a quick demo, you can run Prometheus locally on your machine. Follow the qui
 
 ### Quickstart: Running Prometheus locally
 
+```{admonition} Note
+:class: note
+If you need to change the root temporary directory by using "--temp-dir" in your ray 
+cluster setup, please follow the [manual steps](#optional-manual-running-prometheus-locally) to setup Prometheus locally.
```
Comment (angelinalg) [inline doc/source/cluster/metrics.md:28]:
```suggestion
cluster setup, follow these [manual steps](#optional-manual-running-prometheus-locally) to set up Prometheus locally.
```

File: doc/source/cluster/metrics.md [hunk 5]
```diff
@@ -76,7 +82,7 @@ tar xvfz prometheus-*.tar.gz
 cd prometheus-*
 ```
 
-Ray provides a Prometheus config that works out of the box. After running Ray, you can find the config at `/tmp/ray/session_latest/metrics/prometheus/prometheus.yml`.
+Ray provides a Prometheus config that works out of the box. After running Ray, you can find the config at `/tmp/ray/session_latest/metrics/prometheus/prometheus.yml`. If you specify the `--temp-dir={your_temp_path}` when starting the ray cluster, the config file will be at `{yout_temp_path}/session_latest/metrics/prometheus/prometheus.yml`
```
Comment (angelinalg) [inline doc/source/cluster/metrics.md:85]:
```suggestion
Ray provides a Prometheus config that works out of the box. After running Ray, you can find the config at `/tmp/ray/session_latest/metrics/prometheus/prometheus.yml`. If you specify the `--temp-dir={your_temp_path}` when starting the Ray cluster, the config file is at: `{your_temp_path}/session_latest/metrics/prometheus/prometheus.yml`
```

**Review by angelinalg — APPROVED** [review]
some style nits and a typo. For future reference, please run Vale to get style feedback: https://docs.ray.io/en/latest/ray-contribute/docs.html#how-to-use-vale

**Pull request merged**